numpy/polynomial/_polybase.py

```diff
@@ -283,10 +315,78 @@ def __repr__(self):
         name = self.__class__.__name__
         return f"{name}({coef}, domain={domain}, window={window})"
 
+    def __format__(self, fmt_str):
+        if fmt_str not in ('ascii', 'unicode'):
+            raise TypeError(
+                f"Unsupported format string '{fmt_str}' passed to "
+                f"{self.__class__}.__format__. Valid options are "
+                f"'ascii' and 'unicode'"
+            )
+        if fmt_str == 'ascii':
+            return self._generate_string(self._str_term_ascii)
+        return self._generate_string(self._str_term_unicode)
+
     def __str__(self):
-        coef = str(self.coef)
-        name = self.nickname
-        return f"{name}({coef})"
+        if self._use_unicode:
+            return self._generate_string(self._str_term_unicode)
+        return self._generate_string(self._str_term_ascii)
+
+    def _generate_string(self, term_method):
+        """
+        Generate the full string representation of the polynomial, using
+        ``term_method`` to generate each polynomial term.
+        """
+        # Get configuration for line breaks
+        linewidth = np.get_printoptions().get('linewidth', 75)
+        if linewidth < 1:
+            linewidth = 1
+        out = ""
+        if self.coef[0] < 0:
+            out += "-"
+        out = f"{self.coef[0]} "
+        cur_line = len(out) // linewidth    # For handling linebreaks
+        for i, coef in enumerate(self.coef[1:]):
+            next_term = ""
+            power = str(i + 1)
+            # Polynomial coefficient
+            if coef >= 0:
+                next_term += f"+ {coef}·"
+            else:
+                next_term += f"- {np.abs(coef)}·"
```

Comment:
Cdot is not ascii, this should be conditional on the formatting type too, using `*` for ascii.

Reply:
Yes, this was intentional. The name `_use_unicode` that I chose for the switch is actually misleading, it should probably be `_use_unicode_superscripts`. The cdot is a unicode character, but it is in the Latin 1 range and thus it *should* be safe for all fonts.

I suppose though in the interest of a truly "failsafe" fallback it does make sense to remove all unicode characters. Thanks for the catch.